pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -677,6 +735,29 @@ func getTaskRunsStatus(pr *v1alpha1.PipelineRun, state []*resources.ResolvedPipe
 	return status
 }
 
+func getFinalTaskRunsStatus(pr *v1alpha1.PipelineRun, state []*resources.ResolvedPipelineRunTask) map[string]*v1alpha1.PipelineRunTaskRunStatus {
```

Comment:
good candidate for unit tests? (and maybe being in a separate package?)

Reply:
yup, lacking lot of unit testing, love your proposal of splitting the implementation into separate PRs, can add more unit tests instead of having one huge XX...XL PR 😱